In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('G:\내 드라이브\web_crawling\linkareer2.csv')

In [6]:
df

CompanyName              CompanyCareer                    CompanyDate  \
0       현대엔지니어링                  자산관리 자산설비                        2021 상반기   
1          삼성생명                     지점사무지원                        2021 상반기   
2        한국연구재단                          IT                            전산    
3          삼성전기                 연구개발직 재료개발                        2021 상반기   
4           네이버    [지식베이스] 네이버 영화 서비스 개발 신입  경력사원 모집 - (신입) 네이버 영화 서비스 개발    
...          ...                        ...                            ...   
9018      현대파워텍                       연구개발                        2013 하반기   
9019      웅진코웨이                        마케팅                        2013 하반기   
9020      현대자동차                       영업관리                        2013 상반기   
9021       한국전력                   인턴[송,배전]                        2013 상반기   
9022       한국전력                   인턴[송,배전]                        2013 상반기   

                                              UserInfos   Score  \
0     한양대 ERICA / 기계공학과 /  토익: 855, 토익스피킹: 130/레벨6 /...    3.78   
1     지방사립 / 경제학 /  토익: 720, 토익스피킹: 140/레벨 6 / 사회생활 ...    4.15   
2     지거국 / 컴퓨터 /  토익: 765, 토익스피킹: , 오픽: , 기타: / 사회생...     3.9   
3     경북대 / 전자재료공학과 /  토익: , 토익스피킹: , 오픽: IM2, 기타: 한...    3.48   
4     인하대 / 컴퓨터공학과 /  토익: , 토익스피킹: , 오픽: IM2, 기타: / ...   3.36    
...                                                 ...     ...   
9018                         경희대 / 기계 /  토익 860 / 인턴 1회   3.76    
9019                         부산경상대학교 / 광고마케팅과 /  토익 850    3.8    
9020  상명대학교 / 국제통상학 /  토익 900, HSK 6급 / 해외인턴1회, 중국어 ...    4.3    
9021              지거국 / 전자공학 /  토익 900대 / 음악봉사활동, 멘토링 등    3.7    
9022                지거국 / 전자공학 /  토익 900 / 음악봉사활동, 멘토링등    3.7    

                                                Content  \
0     본인이 회사를 선택할때의 기준은 무엇이며, 왜 현대엔지니어링이 그 기준에 적합한지를...   
1     [금융 백종원] 백종원은 여러 매체를 통해 요리에 관한 각종 정보와 비결을 공유하며...   
2     1. 우리 재단 인턴으로 지원한 가장 큰 동기 및 인턴기간 동안 이루고자하는 목표에...   
3     1. 삼성전기를 지원한 이유와 입사 후 회사에서 이루고 싶은 꿈을 기술하십시오.\n...   
4     자신이 개발자로서 성장 목표와 그 목표를 달성하기 위해 무엇을 할 것인지 기술해 주...   
...                                                 ...   
9018  1. 성장과정\n\n<견학을 통해 얻은 특허>\n\n특허를 취득하고 싶어서 아이디어...   
9019  1. 지원동기\n\n인재와 기업이 함께 성장할 수 있는 곳, 그곳이 바로 웅진코웨이...   
9020  1. 현대자동차에 지원하게 된 동기 또는 해당 직무에 지원하게 된 동기는 무엇입니까...   
9021  [질문 1] 취업이라는 목표 이외에 인생의 큰 목표를 세운 경험, 그 목표를 실현하...   
9022  한전 자가소개서 양식 500자 12줄 이내\n\n[질문 1] 취업이라는 목표 이외에...   

                                                    URL  
0     https://linkareer.com/cover-letter/25204?page=...  
1     https://linkareer.com/cover-letter/25579?page=...  
2     https://linkareer.com/cover-letter/28761?page=...  
3     https://linkareer.com/cover-letter/28759?page=...  
4     https://linkareer.com/cover-letter/28665?page=...  
...                                                 ...  
9018  https://linkareer.com/cover-letter/11797?page=...  
9019  https://linkareer.com/cover-letter/13306?page=...  
9020  https://linkareer.com/cover-letter/20654?page=...  
9021  https://linkareer.com/cover-letter/22089?page=...  
9022  https://linkareer.com/cover-letter/17456?page=...  

[9023 rows x 7 columns]

## [1] Company확인

### [1-1] 회사 확인

* Company Name => 소문자 모두 대문자로 변경하여 같은 회사 합치기

In [8]:
CompanyNames = set(df.CompanyName.unique().tolist())

In [9]:
len(CompanyNames)

1553

### [1-2] Company 정보가 3개 이상

In [5]:
df.loc[4]

CompanyName                                                   네이버 
CompanyCareer                             [지식베이스] 네이버 영화 서비스 개발 신입
CompanyDate                          경력사원 모집 - (신입) 네이버 영화 서비스 개발 
UserInfos        인하대 / 컴퓨터공학과 /  토익: , 토익스피킹: , 오픽: IM2, 기타: / ...
Score                                                        3.36 
Content          자신이 개발자로서 성장 목표와 그 목표를 달성하기 위해 무엇을 할 것인지 기술해 주...
URL              https://linkareer.com/cover-letter/28665?page=...
Name: 4, dtype: object

* 위 예시 처럼 Company 정보는 원래 [회사]/[직군]/[날짜] 형식이어야 되는데 직군을 길게 적을 경우 ("/") split 시 3개 이상 됨
* 이런 데이터 확인해서 변경하기

## [2] User 정보

* user 정보가 아예 없는 경우 있음 => 이 경우 그냥 null로 대체

### [2-1] 학점 정보

* 학점은 4가지 형태가 있을 수 있음

1. 학점 3.5/4.5 인 경우 =>  re.search('학점(.?)\d*(\.?\d*)(./)\d*(\.?\d*)(.)/', user_info)
2. 학점 3.5 / 인 경우 => re.search('학점(.?)\d*(\.?\d*)(./)\d*(\.?\d*)(.)/', user_info)
3. 학점 3.5 인 경우 => re.search('학점(.?)\d*(\.?\d*)', user_info)
4. 학점 정보 없음 => if 학점 in user_info

### [2-2] "/"으로 split해서 정보 확인

* 먼저 어떤 특수문자들이 있는지 파악 필요

In [14]:
users = df.UserInfos.tolist()

In [17]:
import re

In [22]:
special = re.compile(r'[^ A-Za-z0-9가-힣+]')
result = set()
for user in users:
    result |= set(special.findall(user))

In [24]:
json_special = dict()

In [25]:
for r in result:
    if r == '/': continue
    json_special[r] = []
    for user in users:
        if r in user: json_special[r].append(user)


In [26]:
json_special.keys()

dict_keys(['-', 'ㄱ', '<', '）', '‘', 'ㅈ', '③', '(', '多', '.', '·', '（', 'ㅗ', '①', ':', '`', '_', '>', ')', '*', '~', 'ㅏ', '害', "'", '新', 'ㄴ', ']', '&', '\\', '有', '\u3000', '[', ',', '?', '外', '②', '中', '無', '=', '’'])

In [27]:
print(len(result))
print(len(json_special.keys()))

41
40


* "-" : pass
* ㄱ : 데이터 전처리 시 'ㄱ' 제거 <어문계열, 경영하ㅏㄱ -> 경영학과로 바꿔주기 이것만>
* < : "<-이건 근데 자소서 내용에 들어가진 않았네요" 제거
* ) : pass
* ‘’ : "지원하는 회사와 분야(직무)에 대한 지원동기를 자유롭게 기술 하세요" 이 부분 제거, 이 문장 말고는 '으로 변경하기
* ㅈ : 제거
* ③, ① : "01.여러분이 선택한 계열사와 해당 직무를 지원한 동기는 무엇인가요?" 이 부분 제거
* ( : pass
* ㅗ : 토ㅗ익 -> 토익으로 변경
* ":" : /말고, :로 구분 지는 부분 -> 한국사검정시험: , 컴퓨터활용능력: , 기타: 정보보안산업기사 or 토익: 765, 토익스피킹: , 오픽: IM1, 기타: / 사회생활 경험: 네이버 인턴 - 일단 pass
* 多, 有, 新 : 이 단어 제거
* 無, 外: 일단 pass 지우지 x
* 中 : 서강대, 성균관대, 한양대로 변경
* ㄴ : -로 변경
* \\ : 제거
* -> : :으로 변경
* \u3000 : 제거
* = : 제거
* ㅏ : "알바 4ㅏ수" -> "알바 4번"
* *: 및으로 변경
* ` : '으로 변경
* ㅗ : "토ㅗ익 870" -> "토익"으로 변경

In [42]:
df[df['UserInfos'].str.contains('(')]['UserInfos'].tolist()

['서울 / 화학 /  토익 920 / 오픽 IH / 토스 160 / 미국 인턴 12개월 봉사 2회 각 12개월, 7개월 01.여러분이 선택한 계열사와 해당 직무를 지원한 동기는 무엇인가요? ① 선택한 계열사가 아니면 안되는 이유, ② 직무에 관심갖게 된 계기, ③ 본인이 직무를 잘 수행할 수 있는 이유(본인의 강점, 노력, 경험 등에 근거)를 반드시 포함하여 구체적으로 작성해주세요. (1000자) 화학을 전공으로서 배우면배울수록 화학이 자연에 모든 해(害)를']

In [65]:
json_special['ㅗ']

['미디어컨텐츠 /  토ㅗ익 870 / 인턴1, 대외활동 2']

## [3] 본문 내용

In [81]:
content = df.Content.tolist()

In [118]:
[re.sub('\n+') for c in content if '낙낙고취' in c][0]

'1. 본인을 채용하게 되면 포스코케미칼에 영향을 줄 수 있는 긍정적인 측면과 기회에 대해 구체적으로 기술하시오.[최소:900byte][최대:1800byte])\n \n낙낙고취(樂樂高就): 일을 즐기면서 하는 사람은 성취도 크다.\n\n 저는 3학년 여름방학에 실험실 생활을 시작으로 2차전지를 접하게 됐습니다. 제게 주어진 과제는 바이오매스(대나무)를 기반으로 한 음극재 제조였습니다. 평범히 음극재를 제조하던 중 대나무가 어느 정도까지 흑연화가 됐는지 확인하기 위해 분석 장비를 접하게 됐습니다.\n\nXRD를 시작으로 여러 분석 장비를 통해 많은 공을 들인 샘플의 변화를 목격하니 신기했고, 샘플에 대한 애정이 생기기 시작했습니다. 3주 동안 정체를 알 수 없던 이물질 때문에 연구를 진행할 수 없어 고생한 적이 있었습니다. 원인이 볼밀 과정에서 깎여 나온 지르코니아였다는 사실을 발견했을 땐 희열을 느끼기도 했습니다. 그리고 충방전 테스트의 결과를 보기까지 대략 2주의 시간이 소요됐는데, 그 긴 기다림 속에서 인내심을 길렀고, 긴 시간이 걸린 만큼 성취감도 크게 맛봤습니다. \n\n이러한 일들을 겪으면서 연구 활동을 점점 즐기기 시작했고, 제 시간을 샘플에 초점을 맞추어 쓰기 시작했습니다. 제가 그렇게 연구에 몰두할 수 있었던 가장 큰 원동력은 전지에 대한 애정이었습니다. \n\n2차전지 산업은 최근 본격적으로 성장할 단계인 산업이라고 생각됩니다. 이때 전지에 관한 역량과 열정을 갖춘 제가 2차전지 산업에 뛰어든다면, 연구하며 길러낸 저의 역량들과 전지에 대한 애정을 연구지원 직무에 발휘하여 포스코케미칼 셀 제조분석 부문에 있어서 이바지할 수 있는 바가 크다고 생각됩니다.\n\n\n2. 제한된 자원(시간, 비용, 인력 등)에도 불구하고 도전적인 목표를 달성했던 경험을 구체적으로 기술하시오.[최소:900byte][최대:1800byte])\n\n우공이산(愚公移山): 한 가지 일을 끝까지 밀고 나가 성공시킴\n\n 저는 20살에 고려대학교 세종캠퍼스 제어계측공학과에 입

### [3-1] \n 여러개를 1개로 줄이기

In [82]:
import re

content2 = [re.sub("\n+", "\n", c) for c in content]

### [3-2] 본문을 (질문 - 내용)으로 나누기

In [99]:
start_num = []

In [100]:
for con in content2:
    if con[0].isdigit(): # 숫자로 시작
        start_num.append(con)

In [101]:
len(start_num)

4700

In [170]:
# 숫자로 시작하는 경우
# ①, ② : 이런 숫자 모두 1,2,3,로 변경하기
new_contents = []
for con in start_num:
    # [1] \d[. ]?[. ]+(.*?)(\n)+ 사용해서 질문 가져오기
    con = con.replace('①', '1').replace('②', '2').replace('③', '3').replace('④', '4').replace('⑤', '5').replace('⑥', '6')
    con = con.replace('⓵', '1').replace('⓶', '2').replace('⓷', '3')
    qa_dict = dict()
    questions = [q[0] for q in re.findall('\d[. )]?[. )]+(.*?)(\n)+', con)]
    answers = re.sub('\d[. )]?[. )]+(.*?)(\n)+', '###############', con).split("###############")[1:]

    for q, a in zip(questions, answers):
        qa_dict[q] = a.strip()
    new_contents.append(qa_dict)

In [171]:
temp = [idx for idx, c in enumerate(new_contents) if not c]

In [172]:
len(temp)

18

In [176]:
for t in temp:
    print(start_num[t])

⓵본인의 진로를 물류업(또는 지원직무분야)으로 선택한 이유와 그 중 ⓶특히 CJ대한통운에 지원한 동기에 대해 구체적으로 기술해주시 바라며, ⓷입사 후 지원 직무에서의 성장 비전을 구체적으로 작성해주시기 바랍니다.(1000자이내)
 
[열정, 저의 유일한 자산입니다]
 
학부시절 ‘국제 물류와 정보’ 강의에서 채찍효과 모의 실습시간을 가졌습니다. 당시 중간 유통상의 역할을 맡았는데 다른 실습자와 정보가 공유되지 않는 상태였기에 나름의 안전재고를 생각하며 주문을 하였습니다. 하지만 대량의 물품 재고 증가로 이어지는 충격적인 결과를 낳았습니다. 이를 통해 불확실성 아래 수요자와 제조공장, 중간상이 서로간의 소통체계가 없을 경우 물품의 재고가 쌓인다는 결론를 도출했습니다. 이 경험은 물류업에서 정보공유의 필요성을 나타냈고, 효율적인 SCM도입과 고객에게 최적화된 물류서비스를 제공하겠다는 목표를 세우는 계기가 되었습니다.
 
현재 CJ대한통운은 CJ계열사 중 가장 공격적으로 해외진출을 하고 있습니다. 지난 5년간 11개의 해외기업을 인수했고, 올해 5월까지 태국 중앙물류센터를 구축할 예정이기 때문입니다. 자신이 가진 전문성을 끊임없이 연구하고 확장시키는 CJ대한통운의 열정에 저는 매료되었습니다.
 
학부시절 물류관리사를 취득하고 전공수업을 많이 들었지만 ‘물류’라는 단어는 여전히 난해했습니다. 산업의 이해를 돕고자 관세물류아카데미 주관 하에 부산 신 항 견학도 하였지만 역부족이었고 이는 더욱 물류를 이해하고 싶은 갈증을 일으켰습니다. 남북철도연결식이 준비되고 철도를 통한 유럽으로의 물류혁신이 구상되고 있는 지금 신 배송루트가 발달할 수 있고 이는 라스트마일을 더욱 단축시킬 것입니다. 그리고 이는 더욱 전문화된 물류인의 필요성을 외치고 있습니다.
 국내 계약물류, 택배사업, 부산항 TO경험을 통한 역량보유, 끝없는 열정을 가지고 있는 CJ대한통운에서 저의 열정을 녹이고 싶고 더욱 전문성을 갖춰 기업에 기여하는 인재가 되고자 지원합니다.
 
입사 후 2년안에 W&D사업영역